### ISSUE TITLE
method cannot be called on `DynamicImage` due to unsatisfied trait bounds

### ISSUE THREAD

**Issue by shalom-aviv** [ajrcarey/pdfium-render#39]
Hi

I'm newbie in rust )

I want to use this lib in mac os app and handle next error during run.

```
error[E0599]: the method `width` exists for enum `DynamicImage`, but its trait bounds were not satisfied
   --> /Users/.../cargo/git/checkouts/pdfium-render-c0a9ea3e3be0834d/a27b6f9/src/page_objects.rs:435:33
    |
435 |         let image_width = image.width();
    |                                 ^^^^^ method cannot be called on `DynamicImage` due to unsatisfied trait bounds
    |
   ::: /Users/.../.cargo/registry/src/github.com-1ecc6299db9ec823/image-0.23.14/./src/dynimage.rs:43:1
    |
43  | pub enum DynamicImage {
    | ---------------------
    | |
    | doesn't satisfy `DynamicImage: PdfPageObjectCommon`
    | doesn't satisfy `DynamicImage: PdfPageObjectPrivate<'_>`
    |
note: trait bound `DynamicImage: PdfPageObjectPrivate<'_>` was not satisfied
   --> /Users/.../.cargo/git/checkouts/pdfium-render-c0a9ea3e3be0834d/a27b6f9/src/page_object.rs:801:8
    |
799 | impl<'a, T> PdfPageObjectCommon<'a> for T
    |             -----------------------     -
800 | where
801 |     T: PdfPageObjectPrivate<'a>,
    |        ^^^^^^^^^^^^^^^^^^^^^^^^ unsatisfied trait bound introduced here
    = help: items from traits can only be used if the trait is in scope
help: the following trait is implemented but not in scope; perhaps add a `use` for it:
    |
4   | use image::GenericImageView;
    |

```

**Comment by ajrcarey** [comment]
Hi @shalom-aviv,

Based on your provided snippet, I would say you are trying to use the `PdfPageObjectCommon::width()` function when you should be using `DynamicImage::width()`, or vice versa. Look carefully at your trait imports to make sure you're using the correct trait.

Just to be clear, DynamicImage is a separate crate and not part of pdfium-render.

Your use of macOS shouldn't be a problem.

If you can provide more code I may be able to offer more guidance, but my instinct is that this isn't a bug in pdfium-render so I will close the issue shortly unless I hear more from you.

**Comment by ajrcarey** [comment]
PS you're using a very old version of the image crate, try updating to the current version 0.24.3.

**Comment by ajrcarey** [comment]
Yes, that's the problem, you're using a version of image that has different trait definitions. Update your image crate to 0.24.3.

Edited Cargo.toml to constrain required version of image crate to at least 0.24.0.

**Comment by ajrcarey** [comment]
Published 0.7.14 including minimum required version of image 0.24.0. Upgrading your project to pdfium-render 0.7.14 should resolve your problem.

**Comment by shalom-aviv** [comment]
Wou 
Thanks for your assistance!
